src/v/cloud_storage/types.cc

```diff
@@ -262,17 +267,53 @@ size_t anomalies::count_segment_meta_anomaly_type(anomaly_type type) const {
     return static_cast<size_t>(count);
 }
 
+/// Returns number of discarded elements
+template<class T, size_t quota = max_number_of_manifest_anomalies>
+size_t insert_with_size_limit(
+  absl::node_hash_set<T>& self, const absl::node_hash_set<T>& other) {
```

Comment:
I'm wondering if it makes sense for the container to be ordered. Naively I'd expect the container to be evicted from LRU, but instead we're removing random elements to satisfy the limits. In the field this may be surprising and easy to make incorrect assumptions about.

Reply:
I think serde doesn't support ordered set. And I don't want to change the data format much because it's already part of the release and somebody may already turn the feature on. 
Also, this will prioritize new data anyway because we're removing from the 'dest' container first.